src/server/pps/server/api_server.go

```diff
@@ -2641,10 +2590,18 @@ func (a *apiServer) listPipelinePtr(ctx context.Context,
 			}
 			// Get parent commit
 			pachClient := a.env.GetPachClient(ctx)
-			ci, err := pachClient.InspectCommit(ppsconsts.SpecRepo, p.SpecCommit.Branch.Name, p.SpecCommit.ID)
+			// allow reading spec commit regardless of calling user
+			pachClient.SetAuthToken(p.AuthToken)
+			ci, err := pachClient.PfsAPIClient.InspectCommit(
+				pachClient.Ctx(),
+				&pfs.InspectCommitRequest{
+					Commit:     p.SpecCommit,
+					BlockState: pfs.CommitState_STARTED,
+				},
+			)
```

Comment:
I plan to switch `InspectCommit` and some others to just take a commit in a future PR, to eliminate both the verbosity and the potential for error by not scrubbing the grpc error (which also adds a stack trace) or losing auth info from the context

Reply:
FWIW we also want to move away from nested GRPC calls, so maybe we can make this a method call and not an RPC?

Reply:
Oh, of course, whoops